### PR TITLE
feat: Add an option to control intent wait for a started URI

### PR DESCRIPTION
**Pull request by shibupanda**
Added condition to handle waitForLaunch capability to call concerned startUrl function.
Which will fix below issue.
https://github.com/appium/appium/issues/14180
It's depends on below PR
https://github.com/appium/appium-adb/pull/520

### CODE REVIEW

File: lib/commands/general.js [hunk 1]
```diff
@@ -236,10 +236,14 @@ commands.setUrl = async function (url) {
 };
 
 commands.mobileDeepLink = async function (opts = {}) {
-  const {url, package: pkg} = opts;
-  return await this.adb.startUri(url, pkg);
+  const {
+    url,
+    package: pkg
```
Comment (mykola-mokhnach) [inline lib/commands/general.js:241]: Please destruct waitForAppLaunch  explicitly from opts and add a doctring to this method

File: lib/commands/general.js [hunk 2]
```diff
@@ -236,10 +236,14 @@ commands.setUrl = async function (url) {
...
+  } = opts;
+  return await this.adb.startUri(url, pkg, opts);
 };
 
+
```
Comment (mykola-mokhnach) [inline lib/commands/general.js:246]: no need for an empty line

File: package.json [hunk 3]
```diff
@@ -41,7 +41,7 @@
   ],
   "dependencies": {
     "@babel/runtime": "^7.0.0",
-    "appium-adb": "^7.29.0",
+    "appium-adb": "^7.30.0",
```
Comment (mykola-mokhnach) [inline package.json:44]: 👍

File: lib/commands/general.js (hunk 1 above)
Comment (shibupanda) [reply lib/commands/general.js:241]: Is it something like below?
commands.mobileDeepLink = async function (opts = {}) {
  const {
    url,
    package: pkg,
    waitForLaunch: wait,
  } = opts;
  return await this.adb.startUri(url, pkg, wait);
};

Comment (mykola-mokhnach) [reply lib/commands/general.js:241]: something like

```
const {
  url,
  package,
  waitForLaunch,
} = opts;
return await this.adb.startUri(url, package, {waitForLaunch});
```

Comment (shibupanda) [reply lib/commands/general.js:241]: Yes this also worked for me. Let me do it.
And docstring was not there you want me to add that too?

Comment (mykola-mokhnach) [reply lib/commands/general.js:241]: yes, I know. But we want to make it better, don't we?

Comment (shibupanda) [reply lib/commands/general.js]: Very true @mykola-mokhnach.

**Comment by mykola-mokhnach** [comment]
@shibupanda Do you have time to add the docstring for the `mobileDeepLink` method?

**Comment by shibupanda** [comment]
@mykola-mokhnach Sorry for the delay I stuck some where so will add it in this weekend or I can create a different pull request for it.

**Comment by mykola-mokhnach** [comment]
np, lets do it on the weekend

File: lib/commands/general.js [hunk 4]
```diff
@@ -234,7 +234,18 @@ commands.mobileViewportScreenshot = async function () {
 commands.setUrl = async function (url) {
   await this.adb.startUri(url, this.opts.appPackage);
 };
+/**
+ * @typedef {object} DeepLinkOpts
+ * @property {?string} url - The name of URL to start.
```
Comment (mykola-mokhnach) [inline lib/commands/general.js:239]: the type would be `{!string}` since this property is required

File: lib/commands/general.js [hunk 5]
```diff
@@ -234,7 +234,18 @@ commands.mobileViewportScreenshot = async function () {
...
+ * @property {?string} package - The name of the package to start the URI with.
```
Comment (mykola-mokhnach) [inline lib/commands/general.js:240]: same here

File: lib/commands/general.js (hunk 4 above)
Comment (shibupanda) [reply lib/commands/general.js:239]: Oh! Yes.

**Comment by shibupanda** [comment]
This needs to be update [here](https://github.com/appium/appium/blob/17004739b90c18caac5b89c1afbe691a3f8a766b/commands-yml/commands/mobile-command.yml#L90) I guess.

**Comment by shibupanda** [comment]
> This needs to be update [here](https://github.com/appium/appium/blob/17004739b90c18caac5b89c1afbe691a3f8a766b/commands-yml/commands/mobile-command.yml#L90) I guess.

Shall I proposed for the changes?

**Comment by mykola-mokhnach** [comment]
simply create a PR

**Pull request merged**